src/libshadertrap/src/parser.cc

```diff
@@ -254,15 +256,66 @@ bool Parser::ParseCommandAssertEqual() {
               }
               argument_identifier_2 = std::move(token);
               return true;
+            }},
+
+           {Token::Type::kKeywordFormat,
+            [this, &format_entries]() -> bool {
+              while (true) {
+                CommandAssertEqual::FormatEntry::Kind kind;
+                switch (tokenizer_->PeekNextToken()->GetType()) {
+                  case Token::Type::kKeywordSkipBytes:
+                    kind = CommandAssertEqual::FormatEntry::Kind::kSkip;
+                    break;
+                  case Token::Type::kKeywordTypeByte:
+                    kind = CommandAssertEqual::FormatEntry::Kind::kByte;
+                    break;
+                  case Token::Type::kKeywordTypeFloat:
+                    kind = CommandAssertEqual::FormatEntry::Kind::kFloat;
+                    break;
+                  case Token::Type::kKeywordTypeInt:
+                    kind = CommandAssertEqual::FormatEntry::Kind::kInt;
+                    break;
+                  case Token::Type::kKeywordTypeUint: {
+                    kind = CommandAssertEqual::FormatEntry::Kind::kUint;
+                    break;
+                    default:
+                      return true;
+                  }
+                }
+                auto format_start_token = tokenizer_->NextToken();
+                size_t count;
+                auto maybe_count = ParseUint32("count");
+                if (!maybe_count.first) {
+                  return false;
+                }
+                count = maybe_count.second;
+                format_entries.push_back(
+                    {std::move(format_start_token), kind, count});
+              }
```

Comment:
At this point I think it would be good to complain if no `FORMAT` entries were provided, so that someone cannot write `FORMAT` followed by nothing.

So please check whether `format_entires` is empty. If it is, give an error saying that an empty sequence of format entries is not allowed. You can use `start_token` as the token for this (I suppose you'll have to add it to the parameters captured by the lambda.)